magma/backend/mlir/mlir_compiler.py

```diff
@@ -37,6 +39,14 @@ def suffix(self):
         return "mlir"
 
     def _run_passes(self):
+        # We elaborate_tuples before finalizing when statements because they
+        # may cause array expansion (e.g. when we have an array of tuples).  If
+        # we remove the flatten_all_tuples requirement, we may finalize when
+        # earlier, but then we may still need this code path as an option
```

Comment:
```suggestion
        # NOTE(leonardt): We elaborate tuples before finalizing when statements 
        # because they may cause array expansion (e.g. when we have an array of 
        # tuples).  If we remove the flatten_all_tuples requirement, we may     
        # finalize when earlier, but then we may still need this code path as an                                                                               
        # option.                                                               
```